plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/binary/KSAnnotationDescriptorImpl.kt

```diff
@@ -44,6 +44,8 @@ class KSAnnotationDescriptorImpl private constructor(val descriptor: AnnotationD
         KSNameImpl.getCached(descriptor.fqName!!.shortName().asString())
     }
 
+    override val useSiteTarget: AnnotationUseSiteTarget? = null
```

Comment:
Why is this always null? Does that mean the annotations appear at use site? If so, the behaviors would be inconsistent.

Reply:
For annotations loaded from descriptors, since descriptors are loaded from library and in library they should appear at the use-site, after all that's the purpose of a use-site target.

Reply:
Alright. Let's document this some where. It can be quite confusing to users. Just saying, if the library already did model it in this way, I don't feel it necessary to stick strictly with grammar structure.